packages/qwen-code/package.nix

```diff
@@ -1,53 +1,52 @@
 {
   lib,
+  nodejs_20,
   buildNpmPackage,
-  fetchFromGitHub,
+  fetchurl,
   fetchNpmDeps,
+  runCommand,
 }:
-
-buildNpmPackage (finalAttrs: {
+let
   pname = "qwen-code";
-  version = "0.0.6";
-
-  src = fetchFromGitHub {
-    owner = "QwenLM";
-    repo = "qwen-code";
-    rev = "v${finalAttrs.version}";
-    hash = "sha256-s4+1hqdlJh5jOy6uZz608n5DzuBR+v/s+7D85oFwQIY=";
-  };
+  version = "0.0.7";
+  srcHash = "sha256-DEaa4uEiuCIBPAjiHCOgPxvtsNXl+82wMjAnypsf+00=";
+  npmDepsHash = "sha256-PPw1HHAIPsKabYqmKz00jOrYj65OR78Y6WXRYZy/Zn8=";
+
+  src = runCommand "gemini-cli-src-with-lock" { } ''
+    mkdir -p $out
+    tar -xzf ${
+      fetchurl {
+        url = "https://registry.npmjs.org/@qwen-code/qwen-code/-/qwen-code-${version}.tgz";
+        hash = "${srcHash}";
+      }
+    } -C $out --strip-components=1
+    cp ${./package-lock.json} $out/package-lock.json
+  '';
```

Comment:
It's preferable to build from source if we can. Was there a specific reason to switch?